internal/note/note_signer.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2023 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package note
+
+import (
+	"bytes"
+	"crypto/ed25519"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+	"unicode"
+	"unicode/utf8"
+
+	"golang.org/x/mod/sumdb/note"
+)
+
+const (
+	algEd25519              = 1
+	algEd25519CosignatureV1 = 4
```

Comment:
We're starting to get a few of these additional note signature algo consts, is there somewhere we could use to keep track of them?